### DISCUSSION TITLE
How to set floating joints

### DISCUSSION THREAD

**Discussion by lanmo0923** [tesseract-robotics/tesseract_python#45] [Q&A]
Hello,I want to set the floating joints in tesseract_python to control the position of the sphere.
![image](https://github.com/tesseract-robotics/tesseract_python/assets/49888704/77e15fc2-1fb5-41ed-9434-659aa4da02ad)

I tried to use the code and specified in urdf two ways, but can not be achieved. It returned an error：Unsupported joint type for joint.How do I use the floating joint?@johnwason
'''
position=[-0.5, -0.5,-0.5]
sphere_link = Link("sphere_link")
sphere_link_visual = Visual()
sphere_link_visual.geometry = Sphere(0.2)
sphere_link.visual.push_back(sphere_link_visual)
sphere_link_collision = Collision()
sphere_link_collision.geometry = Sphere(0.2)
sphere_link.collision.push_back(sphere_link_collision)
sphere_joint = Joint("joint_4")
sphere_joint.parent_link_name = "base_link"
sphere_joint.child_link_name = sphere_link.getName()
sphere_joint.type = JointType_FLOATING
sphere_link_joint_transform = Isometry3d.Identity() * Translation3d(*position)
sphere_joint.parent_to_joint_origin_transform = sphere_link_joint_transform
add_sphere_command = AddLinkCommand(sphere_link, sphere_joint)
t_env.applyCommand(add_sphere_command)
```

**Comment by johnwason** [comment]
@Levi-Armstrong 

I don't think floating joints are supported. You will have to use prismatic and revolute joints to get the same effect.

**Reply by Levi-Armstrong** [reply]
In case of Tesseract all joints are more or less floating joints since you can modify/replace any joint using environment commands. Now this approach is not ideal if you are wanting to update the joint transform at a high rate because the command history would grow significantly.
